controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -351,6 +358,41 @@ void master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
     LOG(DEBUG) << "sent";
 }
 
+bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
+                                                              ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto tlvAp = cmdu_rx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
+    if (tlvAp == nullptr) {
+        LOG(ERROR) << "Failed to get APRadioBasicCapabilities";
+        return false;
+    }
+
+    auto tlvWscM1 = cmdu_rx.addClass<ieee1905_1::tlvWscM1>();
```

Comment:
How can the controller receive M2?

Reply:
If there is an agent on the same device, the destination AL-MAC is the device itself, so the ieee1905_transport will publish the CMDU back on the local bus. Since the CMDU is of type WSC (not WSC M2), the controller is subscribed to it so it will receive its own CMDU.

Reply:
Note that this is also already handled by looking at the Radio Info / Radio Identifier TLV, so it's OK for now.